hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java

```diff
@@ -150,6 +167,8 @@ public void map(ImmutableBytesWritable row, final Result value, Context context)
       throws IOException {
       if (replicatedScanner == null) {
         Configuration conf = context.getConfiguration();
+        reCompareTries = conf.getInt(NAME + ".recompareTries", 1);
```

Comment:
actually im realizing we might want to make this dynamic -- 

- from the perspective of backwards compatibility, by default there were no retries. so a default value of 0 makes sense.
- previously, the way to specify retries was to set sleepMsBeforeReCompare and it would always just do 1 retry

so i think we should default this to 0, and then below add:

```java
if (sleepMsBeforeReCompare > 0) {
  reCompareTries = Math.max(reCompareTries, 1);
}
```

meaning, we've now aligned the these values to make sense together.  If someone explicitly configures more retries, they'll get them. And otherwise they will get 1 retry if they specified sleepMsBeforeReCompare or 0 if not, like it used to work.

Then, way below where you currently decide whether to create a runnable or not, you should base it on `reCompareTries == 0` rather than `sleepMsBeforeReCompare == 0`.

I think this makes a little more intuitive sense, since what it's basing the logic on whats actually happening -- whether there are retries are not, rather than what the backoff is for those retries. The other benefit is if someone for some reason wanted to try 3 times with zero backoff, currently they can't do that but in the new setup they would.

Reply:
good point, I think those changes make sense!